src/common/client/oauth-api-client/oauth-api-client.ts

```diff
@@ -0,0 +1,74 @@
+import { ApiClient, bugsplatAppHostUrl, BugSplatResponse } from '@common';
+import fetchPonyfill from 'fetch-ponyfill';
+import FormData from 'form-data';
+
+export class OAuthApiClient implements ApiClient {
+
+    private _accessToken = '';
+    private _tokenType = '';
+    private _createFormData = () => new FormData();
+    private _fetch = fetchPonyfill().fetch;
+
+    constructor(
+        private _clientId: string,
+        private _clientSecret: string,
+        private _host: string = bugsplatAppHostUrl
+    ) { }
+
+    static async createAuthenticatedClient(
+        clientId: string,
+        clientSecret: string,
+        host: string = bugsplatAppHostUrl
+    ): Promise<OAuthApiClient> {
+        const client = new OAuthApiClient(
+            clientId,
+            clientSecret,
+            host
+        );
+        await client.login();
+        return client;
+    }
+
+    async login(): Promise<BugSplatResponse> {
+        const url = `${this._host}/oauth2/authorize`;
+        const method = 'POST';
+        const body = this.createFormData();
+        body.append('grant_type', 'client_credentials');
+        body.append('client_id', this._clientId);
+        body.append('client_secret', this._clientSecret);
+        body.append('scope', 'restricted');
```

Comment:
Remove this

Reply:
Leave it, you can request a different scope with the token call https://datatracker.ietf.org/doc/html/rfc6749#section-3.3